ext/rugged/rugged_patch.c

```diff
@@ -344,6 +353,18 @@ static int patch_print_cb(
 
 	rb_ary_push(rb_buffer, rb_str_new(line->content, line->content_len));
 
+	if (print_payload->timeout > 0) {
+		struct timeval finish;
+		gettimeofday(&finish, NULL);
```

Comment:
FWIW, any [time functions in libgit2](https://github.com/libgit2/libgit2/blob/20302aa43738a972e0bd2e2ee6ae479208427b31/src/util.h#L598) use `clock_gettime(CLOCK_MONOTONIC...` when it's available and fall back to `gettimeofday`.  We never use `mach_absolute_time`, so that might be a precedent for you.